### PR TITLE
external sourcemaps

### LINKED ISSUE #112: Don't include sourcemaps in docs and landing distribution builds

### LINKED ISSUE #449: file.ts is not in the SourceMap
### Bug report

- __Package version(s)__: 1.4.1
- __Browser and OS versions__: webpack

#### Steps to reproduce

1. install `@blueprintjs/core`
1. use Webpack `devtool: "source-map"` (other devtools repro as well)

#### Actual behavior

```
Error: "/src/index.ts" is not in the SourceMap.
```

#### Expected behavior

_silent success_

### PR DESCRIPTION
**Pull request by giladgray**
#### Fixes #112, fixes #449 

- [x] test external sourcemaps in a representative app

#### Changes proposed in this pull request:

- externalize sourcemaps from typescript: 
  image
  this results in a nearly 60% decrease in bundle size (~800kB &rarr; ~300kB for `core`)
- simpler compile alias (by adding `icons` task dep to `typescript-compile`)
- fix "file.ts is not in SourceMap" errors by doing the same thing we do for sass sourcemaps
- use source-map-loader in docs for original TS

### CODE REVIEW

**Comment by giladgray** [comment]
How would you propose testing such things? This is an almost trivial change from inline to external sourcemaps, I struggle to believe it _won't_ work.

**Comment by giladgray** [comment]
the sourcemap experience for docs development is unchanged by this PR: all sourcemapped files still show up in the exact same places. 

one caveat is that `core` references `dist/**/*.js` files while the other packages use `src/**/*.tsx?` files. this is because `core` examples import from `@blueprintjs/core` while others use relative paths.

**Comment by giladgray** [comment]
in the interest of time and sanity, i'm going to save this for next year when we can put the time into solving source maps for realsies.

**Comment by giladgray** [comment]
ok i highline'd myself a "representative app" playground and iterated helplessly until I realized we'd [already solved this problem in our sass sourcemaps](https://github.com/palantir/blueprint/blob/master/gulp/sass.js#L85-L86). doing the same thing for typescript immediately resolved the "file.ts is not in the SourceMap" issue :man_facepalming:

then i went and fixed the sourcemaps for the docs site by adding `source-map-loader` 🛠

File: gulp/typescript.js [hunk 1]
```diff
@@ -48,7 +48,7 @@ module.exports = (gulp, plugins, blueprint) => {
 
         // write sourcemaps to .js files; output .js and .d.ts files
         return mergeStream([
-            tsResult.js.pipe(plugins.sourcemaps.write()),
+            tsResult.js.pipe(plugins.sourcemaps.write(".", { sourceRoot: null })),
```
Comment (adidahiya) [inline gulp/typescript.js]: why do we need to supply the sourceRoot option? would be nice to document this

Comment (giladgray) [inline gulp/typescript.js]: the other usage has a link to github comment but i'll duplicate it here

**Pull request merged**